dspace-api/src/main/java/org/dspace/content/PreviewContentServiceImpl.java

```diff
@@ -307,69 +302,113 @@ private Path createTempFile(String fileType) throws IOException {
      */
     private void addFilePath(List<String> filePaths, String path, long size) {
-        String fileInfo = (Files.isDirectory(Paths.get(path))) ? path + "/|" + size : path + "|" + size;
+        String fileInfo = Files.isDirectory(Paths.get(path)) ? path + "/|" + size : path + "|" + size;
         filePaths.add(fileInfo);
     }
 
     /**
-     * Processes a TAR file, extracting its entries and adding their paths to the provided list.
-     * @param filePaths the list to populate with the extracted file paths
-     * @param tempFile  the temporary TAR file to process
+     * Processes a TAR file, extracting its entries and adding their paths and sizes to the provided list.
+     * Utilizes parallelism to process each TAR entry concurrently.
+     *
+     * @param filePaths the list to populate with the extracted file paths and sizes
+     * @param inputStream the InputStream containing the TAR file data
      * @throws IOException if an I/O error occurs while reading the TAR file
+     * @throws InterruptedException if the current thread is interrupted while waiting for the completion of a task
+     * @throws ExecutionException if an exception occurs during the execution of a parallel task
      */
-    private void processTarFile(List<String> filePaths, Path tempFile) throws IOException {
-        try (InputStream fi = Files.newInputStream(tempFile);
-             TarArchiveInputStream tis = new TarArchiveInputStream(fi)) {
+    private void processTarFile(List<String> filePaths, InputStream inputStream)
+            throws IOException, InterruptedException, ExecutionException {
+        List<TarArchiveEntry> entries = new ArrayList<>();
+        int threadPoolSize = (archiveThreadPoolSize != null) ?
+                archiveThreadPoolSize : Runtime.getRuntime().availableProcessors();
+        ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+        List<Future<List<String>>> futures = new ArrayList<>();
+
+        try (BufferedInputStream bufferedStream = new BufferedInputStream(inputStream);
+             TarArchiveInputStream tis = new TarArchiveInputStream(bufferedStream)) {
+
             TarArchiveEntry entry;
             while ((entry = tis.getNextTarEntry()) != null) {
-                addFilePath(filePaths, entry.getName(), entry.getSize());
+                if (!entry.isDirectory()) {
+                    entries.add(entry);
+                }
+            }
+            // Process sequentially if below threshold
+            if (entries.size() < archiveThreadPoolSize) {
+                for (TarArchiveEntry e : entries) {
+                    addFilePath(filePaths, e.getName(), e.getSize());
+                }
+                return;
             }
+            // Process in parallel if above threshold
+            for (TarArchiveEntry e : entries) {
+                String entryName = e.getName();
+                long fileSize = e.getSize();
```

Comment:
Code Rabbit:
ZIP entries can report -1 for size when the size is unknown. The current implementation doesn't account for this special case.

                String entryName = e.getName();
                long fileSize = e.getSize();
+               // Handle unknown size (-1)
+               if (fileSize < 0) {
+                   fileSize = 0; // or some other appropriate default
+               }
                Callable<List<String>> task = () -> {

Reply:
I removed the parallelism. This comment should be unimportant.